src/NuGet.Clients/NuGet.CommandLine/Commands/SearchCommand.cs

```diff
@@ -0,0 +1,174 @@
+            return packageSources;
+        }
+
+        public override async Task ExecuteCommandAsync()
+        {
+            string sourceSeparator = new string('=', _lineSeparatorLength);
+            string packageSeparator = new string('-', _lineSeparatorLength);
+
+            ILogger logger = Console;
+            CancellationToken cancellationToken = CancellationToken.None;
+
+            SearchFilter searchFilter = new SearchFilter(includePrerelease: PreRelease);
+
+            IList<PackageSource> listEndpoints = GetEndpointsAsync();
+
+            foreach (PackageSource source in listEndpoints)
+            {
+                SourceRepository repository = Repository.Factory.GetCoreV3(source);
+                PackageSearchResource resource = await repository.GetResourceAsync<PackageSearchResource>();
+
+                if (resource is null)
+                {
+                    continue;
+                }
+
+                IEnumerable<IPackageSearchMetadata> results = await resource.SearchAsync(
+                    string.Join(" ", Arguments).Trim(), // The arguments are joined with spaces to form a single query string
+                    searchFilter,
+                    skip: 0,
+                    take: Take,
+                    logger,
+                    cancellationToken);
+
+                Console.WriteLine(sourceSeparator);
+                Console.WriteLine($"Source: {source.Name}");
+
+                if (results.Any())
+                {
+                    if (Verbosity == Verbosity.Quiet)
+                    {
+                        System.Console.WriteLine($"Source: {source.Name}"); // Would not have printed using Console.WriteLine if Verbosity == quiet
+                        System.Console.WriteLine(packageSeparator);
+                    }
+
+                    PrintResults(results);
+                }
+                else
+                {
+                    Console.WriteLine(packageSeparator);
+                    Console.WriteLine("No results found.");
+                    Console.WriteLine(packageSeparator + Environment.NewLine);
+                }
+            }
+        }
+
+        private void PrintResults(IEnumerable<IPackageSearchMetadata> results)
+        {
+            string packageSeparator = new string('-', _lineSeparatorLength);
+
+            foreach (IPackageSearchMetadata result in results)
+            {
+                Console.WriteLine(packageSeparator);
+
+                CultureInfo culture = CultureInfo.CurrentCulture;
+
+                StringBuilder content = new StringBuilder();
+                content.Append($"> {result.Identity.Id} | {result.Identity.Version.ToNormalizedString()}"); // Basic info (Name | Version)
+
+                if (Verbosity != Verbosity.Quiet)
+                {
+                    if (result.DownloadCount != null)
+                    {
+                        string downloads = string.Format(culture, "{0:N}", result.DownloadCount);
```

Comment:
Is this a total download count or per version>

Reply:
This is the total count.